### ISSUE TITLE
Songs in library can exist multiple times, even if there is only 1 file for the song.

### ISSUE THREAD

**Issue by Rawrior** [SEbbaDK/sonica#87]
In search:
![image](https://user-images.githubusercontent.com/9444082/155489639-2d5816eb-5a82-47f4-b6b4-7014be1a3958.png)

Actual files: 
![image](https://user-images.githubusercontent.com/9444082/155489733-583baaff-e175-4013-82e9-9b94b052aa0d.png)

**Comment by SEbbaDK** [comment]
Om guessing the three versions were added in the same session, as i dont think this will happen when loading. We could solve it with a set/hashmap keyed on file path?

**Comment by Rawrior** [comment]
> Om guessing the three versions were added in the same session, as i dont think this will happen when loading. We could solve it with a set/hashmap keyed on file path?

That's true, although the downloader does overwrite the files. I guess the simplest solution would just be to check if there's an identical title+artist in the library already.
